internal/exec/util/unit.go

```diff
@@ -110,6 +110,51 @@ func (ut Util) MaskUnit(unit types.Unit) (string, error) {
 	return filepath.Join("/", SystemdUnitsPath(), unit.Name), nil
 }
 
+// UnmaskUnit deletes the symlink to /dev/null for a masked unit
+func (ut Util) UnmaskUnit(unit types.Unit) error {
+	path, err := ut.JoinPath(SystemdUnitsPath(), unit.Name)
+	if err != nil {
+		return err
+	}
+	// Remove the symlink
+	if err = os.Remove(path); err != nil {
+		return err
+	}
+	return nil
+}
+
+// IsUnitMasked returns true/false if a systemd unit is masked
+func (ut Util) IsUnitMasked(unit types.Unit) (bool, error) {
+	path, err := ut.JoinPath(SystemdUnitsPath(), unit.Name)
+	if err != nil {
+		return false, err
+	}
+
+	// If the path doesn't exist then the unit isn't masked
+	fi, err := os.Lstat(path)
```

Comment:
The `Lstat` shouldn't be necessary; we can just `Readlink` and check the return code.

Reply:
If I call readlink on a path that doesn't exist I get a not exists error. If I call it on something that isn't a link I get:

```
CRITICAL : Ignition failed: failed to create units: readlink /sysroot/etc/systemd/system/kola-runext.service: invalid argument
```

You're saying I should ignore all errors from the `Readlink` call and assume the unit isn't masked? Are there other errors we wouldn't want to ignore? I wasn't sure about the last part so I went through verifying it existed and was a link first with the call to `Lstat`.

Reply:
EINVAL is the documented error code when readlink(2) is called on not-a-symlink.  I'm saying we could check for that return code explicitly, rather than separately lstatting the link.

Reply:
ok I think I've handled it now.